packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -57,6 +58,164 @@ function findInputObjectType(input: IntrospectionInputValue, schema: Introspecti
     return filterType;
 }
 
+function getRequiredQueryArgs({ gridQueryType }: { gridQueryType: IntrospectionField }) {
+    const skipParams = ["offset", "limit", "sort", "search", "filter", "scope"];
+    const requiredParams: IntrospectionInputValue[] = [];
+    gridQueryType.args.forEach((arg) => {
+        if (skipParams.includes(arg.name)) return;
+        if (arg.type.kind === "NON_NULL") {
+            requiredParams.push(arg);
+        }
+    });
+    return requiredParams;
+}
+
+function getRequiredMutationArgs({ createMutationType }: { createMutationType: IntrospectionField }) {
+    const skipParams = ["input"];
+    const requiredParams: IntrospectionInputValue[] = [];
+    createMutationType.args.forEach((arg) => {
+        if (skipParams.includes(arg.name)) return;
+        if (arg.type.kind === "NON_NULL") {
+            requiredParams.push(arg);
+        }
+    });
+    return requiredParams;
+}
+
+function getRequiredGqlArgs({
+    config,
```

Comment:
Config isn't used in this function, so it can be removed. You can probably remove it in all functions that call this function.